web/docs/tql2/operators/from.md

```diff
@@ -3,3 +3,241 @@ sidebar_class_name: hidden
 ---
 
 # from
+
+Obtains events from an URI, inferring the source, compression and format.
+
+```tql
+from uri:str, [loader_args… { … }]
+from events…
+```
+
+:::tip Use `from` if you can
+The `from` operator is designed as an easy way to get data into Tenzir,
+without having to manually write the separate steps of data ingestion manually.
+:::
+
+## Description
+
+The `from` operator is an easy way to get data into Tenzir.
+It will try to infer the connector, compression and format based on the given URI.
+
+Alternatively, it can be used to create events from records.
+
+### `uri: str`
+
+The URI to load from.
+
+### `loader_args… (optional)`
+
+An optional set of arguments passed to the loader.
+This can be used to e.g. pass credentials to a connector:
+
+```tql
+from "https://example.org/file.json", headers={Token: "XYZ"}
+```
+
+### `{ … } (optional)`
+
+A pipeline that can be used if inference for the compression or format does not work
+or is not sufficient.
+
+Providing this pipeline
+disables the inference for the decompression and parsing format in order to avoid
+confusion.
+
+:::tip Only specify `{ … }` if you need it for data ingestion
+The `{ … }` argument exists for data ingestion purposes. In most situations,
+inference should be sufficient and the pipeline should not be required.
+
+If you want to perform other operations on the data afterwards, continue the
+pipeline after this operator instead of providing a sub-pipeline.
+:::
+
+### `events…`
+
+Instead of a URI, you can also provide one or more `record`s, which will be
+the operators output. This is mostly useful for testing pipelines without loading
+actual data.
+
+The operator also accepts lists of records. Those records will be merged into a
+single schema.
```

Comment:
I feel like we should pick one of the following three options:
1. Events are always given in a single list.
2. Events are always given as multiple arguments (note that this doesn't support the "from no-events" use-case).
3. There is always a single argument that is either a record or a list of records.

As I wrote above, I feel like mixing all of this seems bad. Would you agree?

Now, whether or not brackets should surround multiple events is probably subjective. Con: It's more to type. Pro: It keeps the door open for things such as `from ["hello", "world"]` for sending a series of strings. Pro: It perhaps communicates more clearly that what follows is a sequence of things. (3) also has the con that going from one to multiple events requires to track back and add the brackets. Thoughts?

Reply:
We had a discussion about this and agreed that mixing is bad. Given that any future upgrade paths to list<string> are very much future and the naming convention suggests that `from` yields events, we decided to go with a variadic record argument and ditch the list syntax entirely.